### DISCUSSION TITLE
Adding a New Header to the Inference Response Header

### DISCUSSION THREAD

**Discussion by BenjaminBraunDev** [triton-inference-server/server#7810] [Q&A] [answered]
I'm trying to add a new header to the response headers when triton is sent an inference request through http via the HTTPAPIServer::InferRequestClass::SetResponseHeader() function in src/http_server.cc, however even something as simple as adding this to the main body doesn't seem to work:

```
// Temp header to check if response header is added correctly
  evhtp_headers_add_header(
      req_->headers_out,
      evhtp_header_new("test-header", "test", 1, 1));
```

I don't see anything new in the response header when I use curl to post a generate request (the actual inference request works, it just doesn't show the header I added):

```
* upload completely sent off: 215 bytes
< HTTP/1.1 200 OK
< Content-Type: application/json
< Transfer-Encoding: chunked
<
{  ...inference response... }
```

I was expecting a new line like `< test-header: test` in the response. Am I trying to add this in the wrong place?

NOTE: It's possible I didn't include my changes properly (I built just the server and copied the /opt/tritonserver/bin and /opt/tritonserver/lib binaries into a different image of tritonserver that had trt_backend so as to not have to build everything at once) and that's why this isn't working, I imagine the changes should be realized in the binaries in bin and lib? If no, where, and if so is SetResponseHeader() the correct place to add this?

**Comment by coolkp** [comment]
@krishung5 Would you happen to know the code pointer where headers in response are added?

**Reply by BenjaminBraunDev** [reply, marked as answer]
Update: SetResponseHeader() is not called in the HandleGenerate() function which handles inference generate requests. Moved logic to there.
